api/valor_api/backend/metrics/llm_call.py

```diff
@@ -0,0 +1,266 @@
+Evaluation Criteria:
+Coherence (1-5) - the collective quality of all sentences. We align this dimension with the DUC quality question of structure and coherence whereby ”the summary should be well-structured and well-organized. The summary should not just be a heap of related information, but should build from sentence to sentence to a coherent body of information about a topic.”
+Evaluation Steps:
+1. Read the text carefully and identify the main topic and key points.
+2. Assign a score for coherence on a scale of 1 to 5, where 1 is the lowest and 5 is the highest based on the Evaluation Criteria. Respond with just the number 1 to 5."""
+
+
+class LLMClient:
+    """
+    Wrapper for calls to an LLM API.
+
+    Parameters
+    ----------
+    api_key : str, optional
+        The API key to use.
+    model_name : str
+        The model to use.
+    """
+
+    api_key: str | None = None
+    model_name: str
+
+    def __init__(
+        self,
+        api_key: str | None = None,
+        model_name: str | None = None,
+    ):
+        """
+        TODO Should we use an __attrs_post_init__ instead?
+        """
+        self.api_key = api_key
+        if model_name is not None:
+            self.model_name = model_name
+
+    def connect(
+        self,
+    ):
+        """
+        Setup the connection to the API.
+        """
+        raise NotImplementedError
+
+    def process_messages(
+        self,
+        messages: list[dict[str, str]],
+    ) -> Any:
+        """
+        All messages should be formatted according to the standard set by OpenAI, and should be modified
+        as needed for other models. This function takes in messages in the OpenAI standard format and converts
+        them to the format required by the model.
+        """
+        return messages
+
+    def __call__(
+        self,
+        messages: list[dict[str, str]],
+    ) -> Any:
+        """
+        Call to the API.
+        """
+        raise NotImplementedError
+
+    def coherence(
+        self,
+        text: str,
+    ) -> int:
+        """
+        Computes coherence, the collective quality of all sentences, for a single piece of text.
+
+        Parameters
+        ----------
+        text: str
+            The text to be evaluated.
+
+        Returns
+        -------
+        int
+            The coherence score will be evaluated as an integer, with 1 indicating the lowest coherence and 5 the highest coherence.
+        """
+        messages = [
+            {"role": "system", "content": COHERENCE_INSTRUCTION},
+            {"role": "user", "content": text},
+        ]
+
+        response = self(messages)
+
+        try:
+            # Valid responses: "5", "\n5", "5\n", "5.", " 5", "5 {explanation}", etc.
+            ret = int(response.strip()[0])
```

Comment:
I'm only taking the first character of the striped string, so it can't be a float.

But still to your point, if the model returns "3.8", then this code would just take the "3" and convert to the integer 3. That doesn't seem ideal. However I instructed the model to respond with an integer, so if it responds with a float, it's misbehaving anyways. Do we care if we are handling that misbehavior in a slightly non-ideal way?